src/main/java/org/osc/controller/nsc/restserver/api/InspectionApis.java

```diff
@@ -0,0 +1,403 @@
+    @PUT
+    public InspectionPortEntity updateInspectionPort(@PathParam("ipAddr") String ipAddr,
+            @PathParam("portId") String portId, InspectionPortEntity entity) throws Exception {
+
+        LOG.info(String.format("Updating the inspection port element id %s ", "" + portId));
+        init(ipAddr);
+        return this.txControl.required(() -> {
+
+            InspectionPortEntity inspectionPortEntity = (InspectionPortEntity) this.api.getInspectionPort(entity);
+
+            if (inspectionPortEntity == null) {
+                String msg = String.format("inspection port element does not exists id: %s\n", entity.getElementId());
+                LOG.error(msg);
+                throw new IllegalArgumentException(msg);
+            }
+            inspectionPortEntity = this.utils.makeInspectionPortEntity(entity);
+            return this.em.merge(inspectionPortEntity);
+        });
+    }
+
+    @Path("/ipAddr/{ipAddr}/insport/{portId}")
+    @DELETE
+    public void deleteInspectionPort(@PathParam("ipAddr") String ipAddr, @PathParam("portId") String portId)
+            throws Exception {
+        if (portId == null) {
+            throw new IllegalArgumentException("Attempt to update null inspection port");
+        }
+        LOG.info(String.format("Deleting the inspection port element for id %s ", "" + portId));
+        init(ipAddr);
+        InspectionPortEntity inspectionPort = new InspectionPortEntity(portId, null, null);
+        this.api.removeInspectionPort(inspectionPort);
+    }
+
+    @Path("/ipAddr/{ipAddr}/insport")
+    @GET
+    public List<String> getInspectionPortIds(@PathParam("ipAddr") String ipAddr) throws Exception {
+        LOG.info("Listing inspection port ids'");
+        init(ipAddr);
+
+        return this.txControl.supports(() -> {
+            List<? extends InspectionPortEntity> elements = this.utils.txInspectionPortEntities();
+            List<String> elementList = new ArrayList<String>();
+            if (elements.isEmpty()) {
+                return elementList;
+            }
+            elementList = elements.stream().map(InspectionPortEntity::getElementId).collect(Collectors.toList());
+            return elementList;
+        });
+    }
+
+    @Path("/ipAddr/{ipAddr}/insport/{insportId}")
+    @GET
+    public InspectionPortEntity getInspectionPortElement(@PathParam("ipAddr") String ipAddr,
+            @PathParam("insportId") String insportId) throws Exception {
+        if (insportId == null) {
+            throw new IllegalArgumentException("Attempt to retrive null inspection port");
+        }
+        LOG.info(String.format("Getting the inspection port element for id %s ", "" + insportId));
+        init(ipAddr);
+        InspectionPortEntity inspectionPort = new InspectionPortEntity(insportId, null, null);
+        return (InspectionPortEntity) this.api.getInspectionPort(inspectionPort);
+    }
+
+    @Path("/ipAddr/{ipAddr}/insport/{insportId}/inshook")
```

Comment:
I dont think you need to make the insphook a child of the insp port. There is a many to one relationship but I dont believe it must be a parent.

Reply:
ok